assets/chezmoi.io/docs/reference/special-files/index.md

```diff
@@ -1,6 +1,41 @@
 # Special files
 
-All files in the source state whose name begins with `.` are ignored by default,
-unless they are one of the special files listed here. `.chezmoidata.$FORMAT` and
-files in `.chezmoidata` folders are read before all other files so that it can
-be used in templates.
+All files in the source directory whose name begins with `.` are ignored by
+default, unless they are one of the special files listed here. All of these
+files are optional and are evaluated in a specific order.
```

Comment:
This order is based on what I believe is correct; I have not verified it in the code. It would be good to get clarity.

Reply:
See

https://github.com/twpayne/chezmoi/blob/9e6d5369ef808968118560b6f2b6a09a8220375d/internal/chezmoi/system.go#L190-L199

and the comment above.